src/Ztm.Data.Entity/Contexts/Main/Block.cs

```diff
@@ -32,27 +32,22 @@ public int CompareTo(Block other)
                 return 1;
             }
 
-            if (Height == other.Height)
+            if (Height != other.Height)
             {
-                if (Hash != null)
-                {
-                    return Hash.CompareTo(other.Hash);
-                }
+                return Height - other.Height;
+            }
 
-                return Hash == other.Hash ? 0 : -1;
+            if (Hash != null)
+            {
+                return Hash.CompareTo(other.Hash);
             }
 
-            return Height - other.Height;
+            return Hash == other.Hash ? 0 : -1;
```

Comment:
We need to modify this every time we add more fields to compare. My idea is change the above `if` to:

```cs
if (Hash != null)
{
    return Hash.CompareTo(other.Hash);
}
else if (other.Hash != null)
{
    return -1;
}
```

Then we can change this line to just `return 0;`.